src/mvba/abba/mod.rs

```diff
@@ -1,30 +1,32 @@
 // TODO: apply section 5.3.3. Further Optimizations
 
+mod error;
 pub(super) mod message;
 
-mod error;
-use self::error::{Error, Result};
-use self::message::{
-    Action, MainVoteAction, MainVoteValue, Message, PreVoteAction, PreVoteJustification,
-    PreVoteValue,
-};
-use super::NodeId;
-use crate::mvba::abba::message::MainVoteJustification;
-use crate::mvba::broadcaster::Broadcaster;
-use blsttc::{PublicKeySet, SecretKeyShare, SignatureShare};
-use log::debug;
 use std::cell::RefCell;
 use std::collections::HashMap;
 use std::rc::Rc;
 
+use blsttc::{PublicKeySet, SecretKeyShare, SignatureShare};
+use log::{debug, error};
+
+use super::NodeId;
+use crate::mvba::abba::message::MainVoteJustification;
+use crate::mvba::broadcaster::Broadcaster;
+use error::{Error, Result};
+use message::{
+    Action, DecisionAction, MainVoteAction, MainVoteValue, Message, PreVoteAction,
+    PreVoteJustification, Value,
+};
+
 pub(crate) const MODULE_NAME: &str = "abba";
 
 /// The ABBA holds the information for Asynchronous Binary Byzantine Agreement protocol.
 pub(crate) struct Abba {
-    id: String,                  // this is same as $ID$ in spec
-    i: NodeId,                   // this is same as $i$ in spec
-    r: usize,                    // this is same as $r$ in spec
-    decided_value: Option<bool>, // TODO: should be boolean?
```

Comment:
I prefer to store boolean here. Because the actual outcome of the ABBA is a boolean, not `Value

Reply:
We need the aggregated signature over the decision so that we can later send it to other nodes as proof of decision.